File: Array/Distribute_candies/scala/Distribute_candies.scala
object candy {
	def main(args:Array[String]){
		println(candiesDistribute(args,args.length))
	}
	def candiesDistribute(candyType:Array[String],n:Int):Int={
		return ((candyType.toSet).size).min(n/2)
	}		
}
object candies {
	def main(args:Array[String]){
		println((args.length/2).min(args.toSet.size))
	}
}

//scala candies 6 6 6 6
//scala candy 6 6 6 6

File: Array/Distribute_candies/javascript/Distribute_candies.js
function candiesDistribute(array,n){
	var candies = new Set();
	array.map(v => candies.add(v));
	return Math.min(candies.size,n/2);
}
console.log(candiesDistribute(process.argv.slice(2),process.argv.slice(2).length))
